### PR TITLE
workshop docs and error handling for `last_fit()`

### LINKED ISSUE #674: More guardrails for input to `last_fit()`
Based on https://tune.tidymodels.org/reference/last_fit.html we should 

- clarify in the documentation that `last_fit()` takes "unfitted" objects as input, so either an unfitted (but finalized) `workflow` or an unfitted model aka a model specification
- throw a more informative error

``` r
library(parsnip)
library(tune)

lm_fit <- linear_reg() %>% fit(mpg ~ ., data = mtcars)

# this should clarify that the object needs to be "unfitted"
last_fit(lm_fit)
#> Error in `last_fit()`:
#> ! The first argument to [last_fit()] should be either a model or workflow.
#> Backtrace:
#>     ▆
#>  1. ├─tune::last_fit(lm_fit)
#>  2. └─tune:::last_fit.default(lm_fit)
#>  3.   └─rlang::abort(msg)
```

<sup>Created on 2023-04-14 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by simonpcouch**
Closes #674. Review may be made easier by viewing each of the commits separately. :)

One tricky bit here: we haven't previously documented that the workflow can't be fitted, and passing a fitted workflow to `last_fit()` _works_:

``` r
library(tidymodels)

lm_fit <- workflow(mpg ~ ., linear_reg()) %>% fit(data = mtcars)

last_fit(lm_fit, initial_split(mtcars))
#> # Resampling results
#> # Manual resampling 
#> # A tibble: 1 × 6
#>   splits         id               .metrics .notes   .predictions     .workflow 
#>   <list>         <chr>            <list>   <list>   <list>           <list>    
#> 1 <split [24/8]> train/test split <tibble> <tibble> <tibble [8 × 4]> <workflow>
```

<sup>Created on 2023-04-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Should we change this? Is that in scope for this PR or should that change live in its own PR?

### CODE REVIEW

File: R/last_fit.R [hunk 1]
```diff
@@ -71,6 +93,16 @@ last_fit.default <- function(object, ...) {
   rlang::abort(msg)
 }
 
+#' @export
+last_fit.model_fit <- function(object, ...) {
+  cli::cli_abort(c(
+    "{.help [{.fun last_fit}](tune::last_fit)} is not well-defined for \\
+     fitted model objects.",
+    "i" = "{.help [{.fun last_fit}](tune::last_fit)} takes \\
+           a model specification or unfitted workflow as its first argument."
```
Comment (simonpcouch) [inline R/last_fit.R:102]: Should "model specification" and "unfitted workflow" also be linked to help-files?

**Review by hfrick — APPROVED** [review]
Nice! 🏄 

Re `last_fit()` working on a fitted workflow: I'd split that into a separate issue and gather some feedback from the team. While I'm surprised it works, I assume the reason for it may be to save you from having to refit your model if you only want a convenient way to get performance metrics for the test set.

**Comment by simonpcouch** [comment]
You got it!

File: R/last_fit.R (hunk 1 above)
Comment (simonpcouch) [reply R/last_fit.R]: You got it. :) Added in [96ce9b3](https://github.com/tidymodels/tune/pull/675/commits/96ce9b3ebc8bd6d074f246a8fd9f2592fdc1a9e6).

**Pull request merged**